src/kaleidoscope/hardware/olkb/Planck.cpp

```diff
@@ -0,0 +1,200 @@
+/* -*- mode: c++ -*-
+ * Kaleidoscope-Hardware-Planck -- OLKB Planck hardware support for Kaleidoscope
+ * Copyright (C) 2018  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of version 3 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef ARDUINO_AVR_PLANCK
+
+#include <Kaleidoscope.h>
+#include <KeyboardioHID.h>
+#include <avr/wdt.h>
+
+static bool do_scan_ = true;
+
+ISR(TIMER1_OVF_vect) {
+  do_scan_ = true;
+}
+
+namespace kaleidoscope {
+namespace hardware {
+namespace olkb {
+
+uint8_t matrix_row_pins[] = { PIN_D0, PIN_D5, PIN_B5, PIN_B6 };
+uint8_t matrix_col_pins[] = { PIN_F1, PIN_F0, PIN_B0, PIN_C7, PIN_F4, PIN_F5, PIN_F6, PIN_F7, PIN_D4, PIN_D6, PIN_B4, PIN_D7 };
+uint8_t matrix_columns = sizeof(matrix_col_pins);
+uint8_t matrix_rows = sizeof(matrix_row_pins);
+
+uint16_t Planck::previousKeyState_[matrix_rows] = {0} ;
+uint16_t Planck::keyState_[matrix_rows] = {0};
```

Comment:
As both of these are static, they are guaranteed to be initialized to zero, and the extra initialization is not needed. We could save a few bytes by not initing them explicitly, I reckon.

Reply:
No change to compiled size, but happy to kill it.